### PR TITLE
Race condition fixed

### PR DESCRIPTION
**Pull request by EvanSmith-git**
Fix for issue #22

### CODE REVIEW

**Review by japrescott — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -53,15 +53,16 @@ function wrapArgs(fns, isParam) {
  */
 
 function wrap(fn, isParam) {
-  const isErrorHandler = !isParam && fn.length == 4;
-  async function wrapped() {
-    next = _once(arguments[2 + isErrorHandler]);
```
Comment (japrescott) [inline index.js:58]: I guess the main problem was that `next` and `res` are not const and therefore global/rewriting itself

**Comment by japrescott** [comment]
@vkarpov15 any feedback to @EvanSmith-git or can this be merged?

**Review by vkarpov15 — APPROVED** [review]
Good find, thanks! It looks like this project has grown sophisticated enough that we need a linter

**Comment by 7opf** [comment]
@vkarpov15 could you please publish a patch for this fix?

**Comment by vkarpov15** [comment]
I just shipped v0.7.2 with the fix, sorry for the delay @7opf  :+1:

**Comment by 7opf** [comment]
No worries, thank you.

**Pull request merged**